### PR TITLE
chore: Improve documentation

### LINKED ISSUE #1021: Improve Documentation
- [ ] Rename extension
- [ ] Revisit README

### PR DESCRIPTION
**Pull request by CsCherrYY**
fix #1021 

This pr introduces:

- Rename extension name to `Gradle for Java`
- Rewrite extension description
- Improve README
  - remove requirement for wrapper
  - add some new features

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
 [![CodeQL](https://github.com/microsoft/vscode-gradle/workflows/CodeQL/badge.svg)](https://github.com/microsoft/vscode-gradle/actions?query=workflow%3ACodeQL)
 [![GitHub bug issues](https://img.shields.io/github/issues/microsoft/vscode-gradle/bug?label=bug%20reports)](https://github.com/microsoft/vscode-gradle/issues?q=is%3Aissue+is%3Aopen+label%3Abug)
 
-This VS Code extension provides a visual interface for your Gradle build. It supports whatever Gradle supports and is language agnostic, but can work nicely alongside other extensions like the [Java language support extension](https://github.com/redhat-developer/vscode-java).
-
-![Gradle Tasks Screencast](images/screencast.gif)
+This VS Code extension provides a visual interface for your Gradle build. You can use this interface to view Gradle Tasks and Project dependencies, or run Gradle Tasks as VS Code Task. The extension also offers better Gradle file (e.g. `build.gradle`) authoring experience including syntax highlighting, error reporting and auto completion. The extension can work nicely alongside other extensions like the [Language Support for Java](https://github.com/redhat-developer/vscode-java).
```
Comment (nickzhums) [inline README.md:9]: The extension can work nicely -> The extension works nicely

Comment (nickzhums) [inline README.md:9]: should we use Expansion Pack instead of Language Support for Java?

File: README.md [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
 ## Requirements
 
 - [VS Code >= 1.60.0](https://code.visualstudio.com/download)
 - [Java >= 8](https://adoptopenjdk.net/)
-- Local [Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) executables
 
 ## Project Discovery
 
-This extension supports both root and nested Gradle projects. A Gradle project is identified by having Gradle Wrapper scripts (`gradlew` or `gradlew.bat`) at the root of a directory. (Nested Gradle project discovery is not enabled by default, set `"gradle.nestedProjects": true` to enable it.)
+This extension supports both root and nested Gradle projects. A Gradle project is identified by having Gradle Wrapper scripts (`gradlew` or `gradlew.bat`) or Gradle build files (`build.gradle` or `settings.gradle` and their kotlin versions) at the root of a directory. (Nested Gradle project discovery is not enabled by default, set `"gradle.nestedProjects": true` to enable it.)
```
Comment (nickzhums) [inline README.md]:
```suggestion
This extension supports both root and nested Gradle projects. A Gradle project is identified by having Gradle Wrapper scripts (`gradlew` or `gradlew.bat`) or Gradle build files (`build.gradle` or `settings.gradle` and their Kotlin versions) at the root of a directory. (Nested Gradle project discovery is not enabled by default, set `"gradle.nestedProjects": true` to enable it.)
```

**Comment by nickzhums** [comment]
we should also update here : 

https://code.visualstudio.com/docs/java/java-build

when we decide to include it in the expansion pack? what's your guys thoughts

File: README.md [hunk 3]
```diff
@@ -161,11 +165,46 @@ Most of the time there should be no reason to stop a daemon. View more info on t
 
 The extension uses the Gradle wrapper to list daemons, and is quite a slow process. If the daemon view is not useful for you, you can simply collapse the view, or disable it completely.
```
Comment (nickzhums) [inline README.md:166]: do we need to provide steps on how to disable? is it easy enough to discover?

File: README.md [hunk 4]
```diff
@@ -161,11 +165,46 @@ Most of the time there should be no reason to stop a daemon. View more info on t
...
+<img src="./images/error.jpg" width="650" alt="Error Reporting" />
+</details>
+<details><summary>Auto completion for Gradle files</summary>
+
+The Gradle language server supports basic auto completions for a Gradle file, including
+- Basic Gradle closures (e.g. dependencies {})
+- Gradle closures from plugins (e.g. java {})
+- Available methods in Gradle closures (e.g. mavenCentral() in dependencies {})
+- Available fields in Gradle closures (e.g. sourceCompatibility in java {})
+- Available dependencies in maven central when declaring a dependency in dependencies closure
```
Comment (nickzhums) [inline README.md]:
```suggestion
- Available dependencies in Maven central when declaring a dependency in dependencies closure
```

File: extension/package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "vscode-gradle",
-  "displayName": "Gradle Tasks",
-  "description": "Run Gradle tasks in VS Code",
+  "displayName": "Gradle for Java",
+  "description": "Manage Gradle Projects, run Gradle tasks and improve Gradle file authoring experience in VS Code",
```
Comment (nickzhums) [inline extension/package.json]:
```suggestion
  "description": "Manage Gradle Projects, run Gradle tasks and provide better Gradle file authoring experience in VS Code",
```

**Comment by nickzhums** [comment]
one more comment: when creating Java projects, should we consider create Gradle project in the future?

**Comment by CsCherrYY** [comment]
> we should also update here :
> 
> https://code.visualstudio.com/docs/java/java-build
> 
> when we decide to include it in the expansion pack? what's your guys thoughts

IMO let's wait for the feedbacks from the next release `3.8.1` to see if there is any blocking issue. And the following changes in my plan:

- Work with @testforstephen to solve debugging related issues https://github.com/microsoft/vscode-gradle/issues?q=is%3Aissue+is%3Aopen+label%3Adebugging
- Add tests for language server
- some valuable enhancement points, such as 
  - https://github.com/microsoft/vscode-gradle/issues/1032
  - https://github.com/microsoft/vscode-gradle/issues/990
  - https://github.com/microsoft/vscode-gradle/issues/963

So maybe after one or two sprints (about Dec. 2021 or Jan. 2022)?

**Comment by CsCherrYY** [comment]
> one more comment: when creating Java projects, should we consider create Gradle project in the future?
> 
> 

It has been tracked in https://github.com/microsoft/vscode-gradle/issues/1027, let's start investigating a good way to implement it.

File: README.md (hunk 1 above)
Comment (CsCherrYY) [reply README.md:9]: 👍, since this extension introduces some debugger features working with Debugger for Java.

File: README.md (hunk 3 above)
Comment (CsCherrYY) [reply README.md:166]: Here it means "disable the daemon view", just right click in the view title and choose `Hide 'Gradle Daemons'`. There will be no getting daemon process if we hide this view. So IMO it's easy to discover.

Comment (nickzhums) [reply README.md:166]: sounds good

**Review by nickzhums — APPROVED** [review]

**Pull request merged**